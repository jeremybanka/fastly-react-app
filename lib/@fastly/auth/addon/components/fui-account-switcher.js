import { OtpSetupError, PasswordRotateError } from '@fastly/auth/utils/errors'
import { set, setProperties } from '@ember/object'
import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'
import { inject as service } from '@ember/service'

export default Component.extend({
  // services
  session: service(),
  store: service(),

  // Internal Properties
  // ---------------------------------------------------------------------------
  accounts: null,
  tagName: '',
  isRequestRunning: false,
  loading: false,
  error: null,

  // Methods
  // ---------------------------------------------------------------------------
  async fetchAccounts() {
    setProperties(this, {
      loading: true,
      didError: false,
    })

    const user = this.session.user

    const accounts = await this.store
      .query('account', { userId: user.id })
      .catch((err) => {
        console.error(err)
        set(this, 'error', 'fetchError')
      })
      .finally(() => {
        set(this, 'loading', false)
      })

    set(this, 'accounts', accounts)
  },
  /**
   * Switch the current session to the chosen account,
   * bailing early if the chosen account is the currentSession
   *
   * @param {Object} account
   * @param {string} account.id
   * @param {boolean} account.currentSession
   */
  handleChooseAccount({ id, currentSession }) {
    if (currentSession) return

    setProperties(this, {
      isRequestRunning: true,
      didError: false,
    })

    return this.session
      .switchAccount(id)
      .catch((err) => {
        if (err instanceof OtpSetupError) {
          set(this, 'error', 'otpSetupError')
        } else if (err instanceof PasswordRotateError) {
          set(this, 'error', 'passwordRotateError')
        } else {
          set(this, 'error', 'authError')
        }
      })
      .finally(() => {
        if (!this.isDestroyed && !this.isDestroying) {
          set(this, 'isRequestRunning', false)
        }
      })
  },

  handleKeyUp(account) {
    const ENTER_KEY_CODE = 13
    if (event.keyCode === ENTER_KEY_CODE) {
      this.handleChooseAccount(account)
    }
  },

  // Ember Hooks
  // ---------------------------------------------------------------------------
  init() {
    this.fetchAccounts()
    this._super(...arguments)
  },

  // Template
  // ---------------------------------------------------------------------------

  layout: hbs`
    <h1 class='fui-auth-screen__header'>
      Switch Accounts
    </h1>

    <p class='margin-bottom--m padding-x--m  text-align--left'>
      Switch to another account linked to {{session.user.login}}.
    </p>

    {{#if (eq error 'authError')}}
      <FuiAuthError />
    {{else if (eq error 'fetchError')}}
      <FuiAuthError
        @errorText='A problem occured while fetching accounts. Try again or'
      />
    {{else if (eq error 'otpSetupError')}}
      <FuiAuthError
        @errorText='2FA is required. To enable this now, visit your'
        @linkText='account settings'
        @link='/account/security'
      />
    {{else if (eq error 'passwordRotateError')}}
      <FuiAuthError
        @errorText='To switch accounts,'
        @linkText='reset your password'
        @link='/account/personal/password'
      />
    {{/if}}

  <ul class='fui-account-switcher__list list--outdented list--unbulleted padding--xs'>
    {{#each accounts as |account|}}
      <li
        tabindex='0'
        role='button'
        aria-current={{if account.currentSession 'true' false}}
        aria-disabled={{if account.currentSession 'true' false}}
        class='fui-account-switcher__list-item fui-card fui-card--accentuated fui-card--focus-ring flex-row flex-row--vcenter padding--m'
        onclick={{action handleChooseAccount account}}
        onkeyup={{action handleKeyUp account}}
      >
        <div class='flex-col--1 text-align--left'>
          <p class='head-bold'>{{account.name}}</p>
          <p class='color--grey'>{{account.id}}</p>
        </div>
        {{#if account.defaultAccount}}
          <div class='font--italic margin-x--l'>Default</div>
        {{/if}}
        <div class="fui-account-switcher__action">
          {{#if account.currentSession}}
            <span class='fui-tag fui-tag--emerald'>Signed In</span>
          {{else}}
            <span class='flex-col--pushed fui-button--link button'>
              Sign In
            </span>
          {{/if}}
        </div>
      </li>
    {{else}}
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
    {{/each}}
  </ul>

  <p class='color--grey padding--s'>
    Don't see the account you're looking for?
    <ExternalLinkTo
      @text='Contact Support'
      @href='https://www.fastly.com/support'
    />.
  </p>
  `,
})
