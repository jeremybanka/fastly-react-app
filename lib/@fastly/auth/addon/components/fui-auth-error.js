import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'

export default Component.extend({
  tagName: '',
  errorText: 'An error occured. Try again or ',
  linkText: 'Contact support',
  link: 'https://www.fastly.com/support',
  class: 'margin-bottom--m',
  dismissible: false,
  layout: hbs`
    <FuiMessage
      @type='error'
      @class='text-align--left {{class}}'
      @dismissible={{dismissible}}
    >
      <p role='alert' ...attributes >
        {{errorText}} <ExternalLinkTo
          @text={{linkText}}
          @href={{link}}
        />.
      </p>
    </FuiMessage>
  `,

  didInsertElement() {
    this._super(...arguments)
    const overlay = document.querySelector('.fui-auth-screen__overlay')
    if (overlay) {
      overlay.scrollTo({
        top: 0,
        behavior: 'smooth',
      })
    }
  },
})
