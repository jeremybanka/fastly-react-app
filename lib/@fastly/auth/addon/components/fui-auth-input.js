import Component from '@ember/component'
import { computed } from '@ember/object'
import { guidFor } from '@ember/object/internals'
import hbs from 'htmlbars-inline-precompile'

export default Component.extend({
  // Passed Properties
  // ---------------------------------------------------------------------------
  label: '',
  // Internal properties
  // ---------------------------------------------------------------------------
  tagName: '',
  inputId: computed(function () {
    return 'input-' + guidFor(this)
  }),

  layout: hbs`
    <div class='fui-auth-screen__input-group' aria-describedby={{tooltipId}}>
      <label class='fui-auth-screen__input-group-label flex-row margin-top--m margin-bottom--s' for={{inputId}}>
        {{label}}
        {{yield (hash contextual-link=(component 'yield'))}}
      </label>
      <Input @class='fui-input authentication__sign-in-input margin-top--s' @value={{value}} ...attributes id={{inputId}} />
    </div>
  `,
})
