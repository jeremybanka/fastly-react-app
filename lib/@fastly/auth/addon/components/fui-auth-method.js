import Component from '@ember/component'
import { computed } from '@ember/object'
import hbs from 'htmlbars-inline-precompile'
import { readOnly } from '@ember/object/computed'
import { inject as service } from '@ember/service'

export default Component.extend({
  tagName: '',
  session: service(),
  store: service(),

  accountName: computed('session.switchCustomerParams.customerId', function () {
    return this.store.peekRecord('account', this.session.switchCustomerParams.customerId).name
  }),

  ssoLoginUrl: readOnly('session.ssoLoginUrl'),
  layout: hbs`
    <h1 class="fui-auth-screen__header">Choose how to sign into <strong>{{accountName}}</strong></h1>
    <div class="padding-x--xxl padding-bottom--l">
      <ExternalLinkTo
        @href={{ssoLoginUrl}}
        class="button primary fui-login-form__button margin-bottom--l"
        aria-label="Sign in with single sign-on"
        target="_self"
      >
        Single sign-on
      </ExternalLinkTo>
      <LinkTo
        @route="switch-customer.login-required"
        class="button new fui-login-form__button margin-left--0"
        aria-label="Sign in with email and password"
      >
        Email and password
      </LinkTo>
    </div>
  `,
})
