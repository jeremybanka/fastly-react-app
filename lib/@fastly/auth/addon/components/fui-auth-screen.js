import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'
import { inject as service } from '@ember/service'

export default Component.extend({
  // Services
  // ---------------------------------------------------------------------------
  router: service(),

  // Internal Properties
  // ---------------------------------------------------------------------------
  tagName: '',

  // Methods
  // ---------------------------------------------------------------------------
  cancel() {
    this.router.location.history.back()
  },

  // Template
  // ---------------------------------------------------------------------------

  layout: hbs`
    <div class='flex-row flex-row--hcenter text-align--center'>
      <div class='fui-card padding--l margin-top--xl'>
        <div class='fui-auth-screen__overlay'>
          <div class='fui-auth-screen__panel'>
            <FuiIcon @name='fastly-logo-red' @class='fastly-logo-red' />
            {{yield}}
            <button class='fui-auth-form__button cancel margin-top--m' {{action cancel}}>Cancel</button>
          </div>
        </div>
      </div>
    </div>
  `,
})
