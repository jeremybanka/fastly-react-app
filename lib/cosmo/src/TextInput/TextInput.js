/* @flow */

import * as React from "react";
import { FormattedMessage } from "react-intl";
import { Box, Flexbox, InputError, InputHelper, InputLabel, Text } from "../";
import styled from "styled-components";
import css from "@styled-system/css";

const StyledInput = styled("input")((props) => {
  return css({
    boxSizing: "border-box",
    outline: "none",
    backgroundColor: "grays.0",

    paddingLeft: "md",
    paddingRight: "md",
    width: "100%",
    height: "xl",
    color: "grays.6",
    fontWeight: "semibold",
    fontSize: "xs",
    letterSpacing: "xs",
    borderRadius: "0",
    borderWidth: "1px",
    borderStyle: "solid",
    borderColor: props.errored ? "reds.5" : "grays.2",
    "&[errored]": {
      borderColor: "reds.5",
    },
    "&:enabled:hover": {
      borderColor: "grays.3",
    },
    "&:enabled:active, &:enabled:focus": {
      borderColor: "grays.6",
    },
    "&:disabled": {
      cursor: "not-allowed", // remove or apply to corresponding label?
    },
    "&:disabled, &[readOnly]": {
      color: "grays.4",
      backgroundColor: "grays.1",
      // unset legacy overrides from Button.css
      opacity: "1",
      pointerEvents: "auto",
    },
    "&::placeholder": {
      color: "grays.4",
      fontWeight: "normal",
    },
  });
});

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type FieldInputProps = {
  name: string,
  id: string,
  value?: string,
  defaultValue?: string,
  placeholder: string,
  minLength?: number,
  maxLength?: number,
  required?: boolean,
  readOnly?: boolean,
  disabled?: boolean,
  type?: "text" | "number",
  onChange?: (e: any) => void,
  onFocus?: (e: any) => void,
  onBlur?: (e: any) => void,
};

type TextInputProps = {
  input: FieldInputProps,
  label?: TextType,
  helper?: TextType,
  error?: ?TextType,
};

const TextInput = (props: TextInputProps) => {
  const { label, helper, error, input } = props;
  const { type = "text" } = input;
  return (
    <Box width="100%">
      {/* if no label is provided, hide all metadata above the input */}
      {label ? (
        <Box marginBottom="xs">
          <Flexbox justifyContent="space-between" alignItems="baseline">
            <Box>
              {/* $FlowFixMe - InputLabel expects text, not array */}
              <InputLabel htmlFor={input.id}>
                {label}
                {input.required ? null : (
                  <Text color="grays.4" marginLeft="xs">
                    (optional)
                  </Text>
                )}
              </InputLabel>
            </Box>
            <Box>
              {input.maxLength ? (
                <Text
                  color="grays.4"
                  fontSize="xxs"
                  as="label"
                  htmlFor={input.id}
                >
                  {input.value.length}/{input.maxLength}
                </Text>
              ) : null}
            </Box>
          </Flexbox>
        </Box>
      ) : null}
      <StyledInput {...input} type={type} errored={!!error} />
      {helper ? (
        <Box marginTop="xs" fontSize="xxs">
          <InputHelper htmlFor={input.id}>{helper}</InputHelper>
        </Box>
      ) : null}
      {error ? (
        <Box marginTop="xs" fontSize="xxs">
          <InputError htmlFor={input.id}>{error}</InputError>
        </Box>
      ) : null}
    </Box>
  );
};

export default TextInput;
