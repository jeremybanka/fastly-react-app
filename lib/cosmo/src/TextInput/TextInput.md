```jsx
<TextInput
  label="So, what are we?"
  input={{}}
  helper="Help me help you"
  error="To err is human, to forgive divine"
/>
```
