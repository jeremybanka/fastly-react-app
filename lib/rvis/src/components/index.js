/* @flow */

import AxisX from "./AxisX";
import AxisY from "./AxisY";
import Background from "./Background";
import Brush from "./Brush";
import BrushActions from "./BrushActions";
import BrushMask from "./BrushMask";
import Chart from "./Chart";
import ChartInteractions from "./ChartInteractions";
import ChartMarker from "./ChartMarker";
import ChartMarkerText from "./ChartMarkerText";
import LegendTotal from "./LegendTotal";
import Loading from "./Loading";

export {
  AxisX,
  AxisY,
  Background,
  Brush,
  BrushActions,
  BrushMask,
  Chart,
  ChartInteractions,
  ChartMarker,
  ChartMarkerText,
  LegendTotal,
  Loading,
};
