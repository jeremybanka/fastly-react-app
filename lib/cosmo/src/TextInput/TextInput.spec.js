/* @flow */

import React from "react";
import { render, shallow } from "enzyme";
import { InputLabel, InputHelper, InputError } from "components";
import TextInput from "./TextInput";

const handleBlur = jest.fn();
const handleChange = jest.fn();
const basicInputObject = {
  name: "email",
  id: "email",
  placeholder: "you@somewhere.com",
  value: "me@here.com", // props.form.values.someName,
  minLength: 3,
  maxLength: 15,
  onChange: handleChange,
  onBlur: handleBlur,
};

describe("TextInput test suite", () => {
  it("renders the stored snapshot", () => {
    const component = (
      <TextInput
        label="Email"
        input={basicInputObject}
        helper="You'll use this to sign in"
        error="Please use a standard email format"
      />
    );
    expect(render(component)).toMatchSnapshot();
  });

  it("renders with InputLabel", () => {
    const component = <TextInput label="Foo" input={basicInputObject} />;
    expect(shallow(component).find(InputLabel)).toHaveLength(1);
  });

  // it('renders without "(optional)" label text', () => {});

  // it('renders with "(optional)" label text', () => {});

  // it("renders character count", () => {});

  it("renders with InputHelper", () => {
    const component = <TextInput helper="Bar" input={basicInputObject} />;
    expect(shallow(component).find(InputHelper)).toHaveLength(1);
  });

  it("renders with InputError", () => {
    const component = <TextInput error="Foo" input={basicInputObject} />;
    expect(shallow(component).find(InputError)).toHaveLength(1);
  });

  it("renders without InputLabel, InputHelper, or InputError", () => {
    const component = <TextInput input={basicInputObject} />;
    expect(shallow(component).find(InputLabel)).toHaveLength(0);
    expect(shallow(component).find(InputHelper)).toHaveLength(0);
    expect(shallow(component).find(InputError)).toHaveLength(0);
  });
});
