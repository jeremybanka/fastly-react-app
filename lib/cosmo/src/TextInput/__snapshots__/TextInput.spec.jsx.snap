// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput test suite renders the stored snapshot 1`] = `
<div
  class="Box-sc-1v2etov-0 dAYJRP"
  width="100%"
>
  <div
    class="Box-sc-1v2etov-0 hKYyow"
  >
    <div
      class="Flexbox-sc-18yd7vq-0 cBxaMO"
    >
      <div
        class="Box-sc-1v2etov-0 kAhoak"
      >
        <label
          class="Text-sc-6c1xd4-0 fktwGx"
          color="grays.6"
          font-size="xs"
          for="email"
          letter-spacing="xs"
        >
          Email
          <span
            class="Text-sc-6c1xd4-0 jWbYly"
            color="grays.4"
          >
            (optional)
          </span>
        </label>
      </div>
      <div
        class="Box-sc-1v2etov-0 kAhoak"
      >
        <label
          class="Text-sc-6c1xd4-0 bDXzpq"
          color="grays.4"
          font-size="xxs"
          for="email"
        >
          11/15
        </label>
      </div>
    </div>
  </div>
  <input
    class="TextInput__StyledInput-za5t3m-0 jAcTdQ"
    id="email"
    maxlength="15"
    minlength="3"
    name="email"
    placeholder="you@somewhere.com"
    type="text"
    value="me@here.com"
  />
  <div
    class="Box-sc-1v2etov-0 dUYKhU"
    font-size="xxs"
  >
    <label
      class="Text-sc-6c1xd4-0 bDXzpq"
      color="grays.4"
      font-size="xxs"
      for="email"
    >
      You'll use this to sign in
    </label>
  </div>
  <div
    class="Box-sc-1v2etov-0 dUYKhU"
    font-size="xxs"
  >
    <label
      class="Text-sc-6c1xd4-0 frfOcY"
      color="reds.5"
      font-size="xxs"
      for="email"
    >
      Please use a standard email format
    </label>
  </div>
</div>
`;
